tooling/minitest/src/tests/locks.rs

```diff
@@ -73,40 +81,41 @@ fn lock_handover() {
 /// If a handover occurs and data race detection does not synchronize the acquirer,
 /// it immediatly writing to global_1 would be a data race with the release.
 fn lock_handover_data_race() {
-    let locals = [<()>::get_type()];
-
-    let ptr_ty = <*const u32>::get_type();
-
-    let p_ptype = <u32>::get_type();
-
-    let b0 = block!(lock_acquire(load(global::<u32>(0)), 1));
-    let b1 = block!(atomic_store(
-        addr_of(global::<*const u32>(1), <*const *const u32>::get_type()),
-        addr_of(global::<u32>(0), ptr_ty),
-        2
-    ));
-    let b2 = block!(lock_release(load(deref(load(global::<*const u32>(1)), p_ptype)), 3));
-    let b3 = block!(return_());
-    let critical = function(Ret::Yes, 0, &locals, &[b0, b1, b2, b3]);
-
-    let locals = [<u32>::get_type(), <()>::get_type()];
-
-    let b0 = block!(storage_live(0), storage_live(1), lock_create(global::<u32>(0), 1),);
-    let b1 = block!(spawn(fn_ptr(1), null(), local(0), 2));
-    let b2 = block!(call(2, &[], local(1), Some(3)));
-    let b3 = block!(join(load(local(0)), 4));
-    let b4 = block!(exit());
-    let main = function(Ret::No, 0, &locals, &[b0, b1, b2, b3, b4]);
-
-    let locals = [<()>::get_type(), <*const ()>::get_type()];
-
-    let b0 = block!(call(2, &[], local(0), Some(1)));
-    let b1 = block!(return_());
-    let second = function(Ret::Yes, 1, &locals, &[b0, b1]);
-
-    let globals = [global_int::<u32>(), global_ptr::<u32>()];
-
-    let p = program_with_globals(&[main, second, critical], &globals);
+    let mut p = ProgramBuilder::new();
+    let lock = p.global_int::<u32>();
+    let gstore = p.global_ptr::<u32>();
+    let mut critical = p.function();
+    critical.ret::<()>();
+    critical.lock_acquire(load(lock));
+    critical.atomic_store(
+        addr_of(gstore, <*const *const u32>::get_type()),
+        addr_of(lock, <*const u32>::get_type()),
+    );
+    critical.lock_release(load(deref(load(gstore), <u32>::get_type())));
+    critical.return_();
+    let critical = p.finish_function(critical);
+
+    let mut second = p.function();
+
+    let mut main = p.function();
+    let thread_id = main.local::<u32>();
+    let ret_place = main.local::<()>();
+    main.storage_live(thread_id);
+    main.storage_live(ret_place);
+    main.lock_create(lock);
+    main.spawn(second.get_function_name(), null(), thread_id);
+    main.call(critical, &[], ret_place, Ret::Yes);
+    main.join(load(thread_id));
+    main.exit();
+    let main = p.finish_function(main);
+
+    let ret = second.ret::<()>();
+    second.arg::<*const ()>();
+    second.call(critical, &[], ret, Ret::Yes);
+    second.return_();
+    p.finish_function(second);
+    p.set_start(main);
```

Comment:
Please have empty lines before and after each function, to make it easier to spot where they begin/end (applies in all the tests)